install/kubernetes/helm/istio/values.yaml

```diff
@@ -191,6 +191,9 @@ global:
   # distributed through the SecretDiscoveryService instead of using K8S secrets to mount the certificates.
   sdsEnabled: false
 
+  # indicate the domain used in SPIFFE identity URL
+  identityDomain: cluster.local 
```

Comment:
This doesn't allow dynamic setting of identityDomain (and we do have emerging use cases that requires this). Please consider configuring the identityDomain in ConfigMap, and refer to the value in ConfigMap as the environmental variable for each deployment.
https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#define-container-environment-variables-using-configmap-data
The benefits are:
1. You have a centralized place to track the identityDomain of the cluster.
2. You don't need to reconfigure every single deployment when the idenityDomain changes during migration.

I suggest we have an "identityDomain" key in "istio" ConfigMap.
@elevran @incfly @wattli

Reply:
@costinm

Reply:
+1 on not making it pilot flag. We'll have to ensure the consistency among multiple components. 

I guess it's fine to put in `ConfigMap` for now. In long term, should we look into integrating with MCP?

@ozevren This config is used to set the SPIFFE uri in the certificate and ideally we want to have the same value across multiple clusters, within a single mesh. Is there any recommendation on this?

Reply:
@myidpt the helm value was good enough for the original PR #7849 (see [diff] (https://github.com/istio/istio/pull/7849/files#diff-7dad29cba9d2ca3e8570c8f65f4b7e86R390)). Maybe it is good enough for this one as well? The current master is kind of broken (any other value for `security.identityDomain` than `cluster.local` will just not work) and this is a step to fix that. Let's track additional work in #8661. WDYT?

Reply:
SGTM, thanks.